api/app/controllers/spree/api/v2/base_controller.rb

```diff
@@ -32,15 +32,25 @@ def spree_current_order
         end
 
         def find_spree_current_order
-          Spree::Order::FindCurrent.new.execute(user: spree_current_user,
-                                                store: spree_current_store,
-                                                guest_token: order_token,
-                                                currency: params[:currency] || current_currency)
+          Spree::Order::FindCurrent.call(
+            guest_token: order_token,
+            currency:    params[:currency] || current_currency,
+            store:       spree_current_store,
+            user:        spree_current_user
+          )
         end
 
         def current_currency
           spree_current_store.default_currency || Spree::Config[:currency]
         end
+
+        def paginate(collection)
```

Comment:
I think this is overkill for pagination. I think that adding `page(params[:page]).per(params[:per_page])` won't be so bad for us. If you have anything "bigger" in mind for this, then I'll be glad to read about the idea

Reply:
Yes, I've thought about it as well but there are 2 points I'd like to consider

1. I don't think that finders should be responsible for pagination. It's simply not their responsibility. And even if it is - passing `page` & `per_page` params & handling pagination logic everywhere looks like a bad code to me due to the duplication. 
Also - I don't think that we should put this code in the controllers - they should be lean HTTP endpoints therefore I'd like to avoid even things like `ModelName.find` there

2. I would like to have a `Paginator` that would be able to paginate every collection so there would be no need to write `page.per` everywhere. Currently it has the simplest implementation but we'll also have to consider fallbacks such as `params[:page] || 1`, `params[:per_page] || Spree::Config.pluralized_model_name_per_page` and so on. It's hard to tell whether we'll need any additional logic at this point of time but IMO it's not that bad - we've centralized pagination logic in one place & `paginate` describes this concepts much better than `page.per`

What do you think about it?